src/loc2d_ros.cpp

```diff
@@ -140,6 +148,21 @@ lama::Loc2DROS::~Loc2DROS()
     delete tfb_;
 }
 
+void lama::Loc2DROS::publishCurrentPose(bool init)
+{
+    if (init)
+    {
+        current_pose_ = loc2d_.getPose();
+        current_orientation_ = tf::createQuaternionFromYaw(current_pose_.rotation());
+    }
+    cur_pose_msg_.pose.pose.position.x = current_pose_.x();
+    cur_pose_msg_.pose.pose.position.y = current_pose_.y();
+    cur_pose_msg_.pose.pose.orientation.z = current_orientation_.getZ();
+    cur_pose_msg_.pose.pose.orientation.w = current_orientation_.getW();
```

Comment:
This is not the correct way the update the orientation of the pose. It should be done with an appropriate function:
```c++
tf::quaternionTFToMsg(tf::createQuaternionFromYaw(current_pose.rotation()), cur_pose_msg.pose.pose.orientation);
```
This will create the propriate quaternion from a *yaw*.

Reply:
I kept the `current_orientation_` variable like I explained above to avoid computing the quaternion twice by calling `tf::createQuaternionFromYaw(double yaw)`